test/functional/element_creation.js

```diff
@@ -69,5 +69,67 @@ describe('element creation', () => {
     var el = container.childNodes[0];
     expect(el.id).to.equal('test');
   });
+
+  describe('for HTML elements', () => {
+    it('should use the xhtml namespace', () => {
+      patch(container, () => {
+        elementVoid('div');
+      });
+
+      var el = container.childNodes[0];
+      expect(el.namespaceURI).to.equal('http://www.w3.org/1999/xhtml');
```

Comment:
I'm a bit confused as to how this actually happens, but please lets not always use the NS method, but instead start out at empty namespace and only switch into the special case where needed.